tests/test_tools.py

```diff
@@ -58,6 +58,7 @@ def many_edge_cases(
     enum: StubEnum,
     defaulted_str: str = "default",
     defaulted_float: float = 1.0,
+    structured_arg: str = "structured",
```

Comment:
Structured implies it's a `BaseModel` not a string, maybe rename to `specified_str`